homeassistant/components/climate/mitsubishicontroller.py

```diff
@@ -0,0 +1,159 @@
+                                              SUPPORT_TARGET_TEMPERATURE,
+                                              SUPPORT_FAN_MODE,
+                                              SUPPORT_OPERATION_MODE,
+                                              SUPPORT_SWING_MODE)
+from homeassistant.const import CONF_URL, ATTR_TEMPERATURE
+from homeassistant.const import TEMP_FAHRENHEIT
+
+REQUIREMENTS = ['mitsPy==0.1.9']
+
+PLATFORM_SCHEMA = PLATFORM_SCHEMA.extend({
+    vol.Required(CONF_URL): cv.string
+})
+
+
+async def async_setup_platform(hass, config, async_add_devices,
+                               discovery_info=None):
+    """Setup platform asynchronously."""
+    from mitsPy.manager import Manager
+    controller = Manager(config.get(CONF_URL))
+
+    async def async_register_devices(raw_devices):
+        """Register devices."""
+        async_add_devices(
+            [MitsubishiHvacDevice(
+                device=device,
+                unit_of_measurement=TEMP_FAHRENHEIT) for device in raw_devices]
+        )
+
+    await controller.initialize(async_register_devices)
+
+
+SUPPORT_FLAGS = (
+    SUPPORT_TARGET_TEMPERATURE
+    | SUPPORT_OPERATION_MODE
+    | SUPPORT_FAN_MODE
+    | SUPPORT_SWING_MODE
+)
+
+
+class MitsubishiHvacDevice(ClimateDevice):
+    """Mitsubishi HVAC Device."""
+
+    def __init__(self, device, unit_of_measurement=None,
+                 current_fan_mode=None):
+        """Initialize device."""
+        self._device = device
+        self._name = self._device.group_name
+        self._current_swing_mode = self._device.current_air_direction
+        self._unit_of_measurement = unit_of_measurement
+        self._current_fan_mode = current_fan_mode
+        self._device.refresh(self.schedule_update_ha_state)
+
+    async def _async_refresh(self):
+        """Refresh function."""
+        await self._device.refresh(lambda: 0)
+
+    @property
+    def should_poll(self):
+        """Should poll."""
+        return True
+
+    @property
+    def name(self):
+        """Device name."""
+        return self._name
+
+    @property
+    def unit_of_measurement(self):
+        """Unit of measurement."""
+        return self._unit_of_measurement
+
+    @property
+    def temperature_unit(self):
+        """Temperature unit."""
+        return self._unit_of_measurement
+
+    @property
+    def current_temperature(self):
+        """Current temperature."""
+        return float(self._device.current_temp_f)
+
+    @property
+    def target_temperature(self):
+        """Target temperature."""
+        return float(self._device.set_temp_value_f)
+
+    @property
+    def state(self):
+        """Current state."""
+        return self.current_operation
+
+    @property
+    def current_operation(self):
+        """Current operation mode."""
+        return self._device.current_operation
+
+    @property
+    def operation_list(self):
+        """List of supported operations."""
+        return self._device.operation_list
```

Comment:
This list should only contain operation modes defined as constants in the base climate component. So import the relevant modes from the climate component and build 1-2 dicts that map between device modes and home assistant modes. Use the dicts when needed to translate between device mode and home assistant mode.